src/src/PluginStructs/P177_data_struct.cpp

```diff
@@ -111,8 +110,12 @@ bool P177_data_struct::plugin_read(struct EventStruct *event) {
       }
     }
 
-    if (tmpTemperature & 0x400) {
-      _temperature = -1.0 * (tmpTemperature & 0x3FF);
+    if (_rawTemperature & 0x8000) {
+      _temperature = -1.0 * (_rawTemperature & 0x7FFF);
+    }
```

Comment:
Maybe it is better to just subtract 65536 when the temperature is >= 32768.

Also a bit odd you use `0x3FF` as mask a few lines up. This way you will never get over `0x8000`.

Just threat it as 2-complement and don't try to use bitwise operations on it as that's quite error prone.